nym-vpn-core/crates/nym-vpn-store/src/key_store.rs

```diff
@@ -0,0 +1,16 @@
+// Copyright 2024 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-3.0-only
+
+use std::error::Error;
+
+use crate::DeviceKeys;
+
+pub trait KeyStore {
+    type StorageError: Error;
+
+    #[allow(async_fn_in_trait)]
```

Comment:
to remove this (assuming we don't export the trait), we could reduce the visibility of the trait to `pub(crate) trait KeyStore`. 

also, it's nice to not have to use `#[async_trait]` everywhere isn't it? : )

Reply:
Unfortunately I think we'll end up using this outside this crate, probably nym-vpn-lib :(